crates/forge/bin/cmd/test/mod.rs

```diff
@@ -814,6 +828,50 @@ fn persist_run_failures(config: &Config, outcome: &TestOutcome) {
     }
 }
 
+/// Generate test report in JUnit XML report format.
+fn junit_xml_report(results: &BTreeMap<String, SuiteResult>, verbosity: u8) -> Report {
+    let mut total_duration = Duration::default();
+    let mut junit_report = Report::new("Test run");
+    junit_report.set_timestamp(Utc::now());
+    for (suite_name, suite_result) in results {
+        let mut test_suite = TestSuite::new(suite_name);
+        total_duration += suite_result.duration;
+        test_suite.set_time(suite_result.duration);
+        test_suite.set_system_out(suite_result.summary());
+        for (test_name, test_result) in &suite_result.test_results {
+            let test_status = if test_result.status.is_failure() {
+                let mut status = TestCaseStatus::non_success(NonSuccessKind::Failure);
+                status.set_message(test_result.reason.clone().unwrap_or_default());
```

Comment:
Can move this `set_message` call to below this if chain since after https://github.com/foundry-rs/foundry/pull/8858 skips can have messages; should also add a test for this

Reply:
added in https://github.com/foundry-rs/foundry/pull/8852/commits/bfe2305812ba125811aff0ed6b7c588e6999ea1c